app/views/samples_reports/show.haml

```diff
@@ -294,14 +294,39 @@
     document.querySelector(`.nih-report-type-modal-container`).classList.add("hidden");
   }
 
-  function downloadNIHFiles() {
+  async function downloadNIHFiles() {
     hideConfirmModal();
-    
-    // Open the download_nih_tables path in a new tab
     var context = getUrlParameter('context');
-    var url = '/samples_reports/'+#{@samples_report.id}+'/nih_tables?context='+context+"";
-    window.open(url, '_blank');
-
+    
+    // if purpose is challenge, then we need the live threshold calculations
+    if ("#{@purpose}" == "Challenge") {
+      var threshold = document.getElementById("threshold").value;
+      var url = '/samples_reports/update_threshold?context='+context+"&samples_report_id="+#{@samples_report.id}+"&threshold="+threshold;
+      await fetch(url)
+          .then((response) => response.json())
+          .then((ret) => {
+            var params = {};
+            params["true_positives"] = ret.confusion_matrix.true_positive || 0;
+            params["true_negatives"] = ret.confusion_matrix.true_negative || 0;
+            params["false_positives"] = ret.confusion_matrix.false_positive || 0;
+            params["tpr"] = (ret.confusion_matrix.true_positive || 0) / ((ret.confusion_matrix.true_positive || 0) + (ret.confusion_matrix.false_negative || 0));
+            params["fpr"] = (ret.confusion_matrix.false_positive || 0) / ((ret.confusion_matrix.false_positive || 0) + (ret.confusion_matrix.true_negative || 0));
+            params["auc"] = #{auc(roc_curve(@samples_report))};
+            params["roc_x_values"] = #{roc_curve(@samples_report).map{|x| x[0]}};
+            params["roc_y_values"] = #{roc_curve(@samples_report).map{|x| x[1]}};
```

Comment:
Didn't found a way to call rails `roc_cuve` and store it in javascript to don't repeat these calls.